packages/regl-worldview/src/commands/GLText.js

```diff
@@ -0,0 +1,393 @@
+// @flow
+
+import TinySDF from "@mapbox/tiny-sdf";
+import memoizeOne from "memoize-one";
+import React, { useState } from "react";
+
+import { defaultBlend, defaultDepth } from "../utils/commandUtils";
+import Command, { type CommonCommandProps } from "./Command";
+import { isColorDark, type TextMarker } from "./Text";
+
+// The GLText command renders text from a Signed Distance Field texture.
+// There are many external resources about SDFs and text rendering in WebGL, including:
+// https://steamcdn-a.akamaihd.net/apps/valve/2007/SIGGRAPH2007_AlphaTestedMagnification.pdf
+// https://blog.mapbox.com/drawing-text-with-signed-distance-fields-in-mapbox-gl-b0933af6f817
+// http://hack.chrons.me/opengl-text-rendering/
+// https://stackoverflow.com/questions/25956272/better-quality-text-in-webgl
+//
+// Approach
+// ========
+// Characters from the font are measured using a <canvas> and the SDFs are drawn into a texture up front
+// (and whenever new characters are being rendered). Then one instanced draw call is made with an instance
+// per character which reads from the corresponding place in the texture atlas.
+//
+// Possible future improvements
+// ============================
+// - Add hitmap support.
+// - Allow customization of font style, maybe highlight ranges.
+// - Add a scaleInvariant option.
+// - Consider a solid rectangular background instead of an outline. This is challenging because the
+//   instances currently overlap, so there will be z-fighting, but might be possible using the stencil buffer and multiple draw calls.
```

Comment:
if you used a fixed width font would instances still overlap?

Reply:
Yeah because there is a radius + buffer around each character in the sdf which you can see on https://mapbox.github.io/tiny-sdf/. I think these demos don't have any z-fighting because they simply don't use the depth buffer